File: LeetCode/Easy/0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        occurred = {}
        for i in range(len(nums)):
            if nums[i] in occurred:
                return [occurred[nums[i]], i]
            else:
                occurred[target - nums[i]] = i
        
        return []

File: LeetCode/Easy/0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    seen := make(map[int]int)

    for idx, n := range nums {
        if s_idx, ok := seen[target - n]; ok {
            return []int{idx, s_idx}
        } else {
            seen[n] = idx
        }
    }
    panic("Unreachable code")
}